server/annotation/middleware.ts

```diff
@@ -0,0 +1,102 @@
+import type { Request, Response, NextFunction } from "express";
+import { isValidObjectId } from "mongoose";
+import AnnotationCollection from "./collection";
+import AnnotationModel from "./model";
+
+/**
+ * Checks if the lecture ID is a valid lecture
+ */
+const isValidLectureId = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const validLecture = req.query.lectureId
+    ? isValidObjectId(req.query.lectureId)
+    : isValidObjectId(req.body.lectureId);
+  const lectureId = req.query.lectureId
+    ? req.query.lectureId
+    : req.body.lectureId;
+  console.log(validLecture, lectureId);
+  // TODO: implement when LectureModel is added.
+  // const annotation = validLecture ? await LectureModel.findById(lectureId) : "";
+  // if (!annotation) {
+  //   res.status(404).json({
+  //     error: "Lecture ID is not valid",
+  //   });
+  //   return;
+  // }
+  next();
+};
+
+/**
+ * Checks if the annotation ID is a valid annotation
+ */
+const isValidAnnotationId = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+) => {
+  const validAnnotation = req.body.annotationId ? req.body.annotationId : req.params.annotationId;
```

Comment:
Add a line above here that says something like:

isValid = req.body.annotationId ? isValidObjectId(req.body.annotationId) : isValidObjectId(req.params.annotationId)

Since without this, if you pass in something that cannot be cast to an ObjectId this will crash the server

Reply:
yup ty